### DISCUSSION TITLE
How to Make Pydantic  Validate List Elements When Using Annotated with Python 3.7

### DISCUSSION THREAD

**Discussion by TonyYanOnFire** [pydantic/pydantic#8675] [Question]
Hello,

I'm encountering an issue with Pydantic's validation in combination with the Annotated type from typing_extensions in Python 3.7. Specifically, it seems that the validation rules are not being applied to the elements within a list.

Here's a minimal example to illustrate the issue:

```
from typing import List
from typing_extensions import Annotated
from pydantic import BaseModel, Field

class Foo(BaseModel):
    bar: Annotated[float, Field(strict=True, gt=0)]
    a: List[Annotated[float, Field(strict=True, gt=0)]] = []

foo_instance = Foo(bar=1.1, a=[-1.1])  # Expected a validation error for `a`
```

In this example, I would expect a validation error when assigning [-1.1] to a, as the elements of the list should be constrained to be greater than 0. However, no validation error is raised.

I'm not sure if I'm missing something in my implementation, or if this is a known issue with Pydantic when used with Python 3.7 and typing_extensions.

Any insights or suggestions would be greatly appreciated.

Thank you!

**Comment by TonyYanOnFire** [comment]
Never mind. I found it. https://docs.pydantic.dev/latest/api/fields/#pydantic.fields.Field

**Reply by jgadling** [reply]
I know you solved your problem, but it's not super clear from the docs how to accomplish this -- what does the solution look like?

**Reply by TonyYanOnFire** [reply]
I kind of forgot. For this case I feel like using [confloat](https://docs.pydantic.dev/latest/api/types/#pydantic.types.confloat)
